### DISCUSSION TITLE
Multichain support for Walletconnect v2

### DISCUSSION THREAD

**Discussion by glitch-txs** [wevm/viem#184] [Idea / Feature Request] [answered]
Walletconnect v2 now allows you to connect a wallet to multiple chains at the same time, and then to make tx request and choose in the request which chain is intended to. This is not possible when wrapping the Walletconnect universal provider with (any library ethersjs/viem)

Here's an issue on this topic where it's explained bettter
https://github.com/WalletConnect/walletconnect-monorepo/issues/1871

Would it be possible to add the chain id to the contract instantiation an maybe if there's not WC v2 provider throw an error when user is not in said chain (for writing calls) otherwise switch the WC provider to the said chain id?

**Comment by jxom** [comment, marked as answer]
viem strictly follows EIP standards, which is single chain by nature. 

[wagmi](https://wagmi.sh/) on the other hand, is multi-chain first, and has first class support for WalletConnect v2. I'd suggest to use [`wagmi`](https://wagmi.sh/) (React) or `@wagmi/core` (Vanilla TS/JS) if you would like multi-chain abstraction.

**Comment by Nilbristi** [comment]


**Reply by tmm** [reply]
This question is better asked in the wagmi discussions area

https://github.com/wagmi-dev/wagmi/discussions
